### DISCUSSION TITLE
DGS Plugin

### DISCUSSION THREAD

**Discussion by RIP21** [dotansimha/graphql-code-generator#6158] [Ideas]
Hey, Netflix recently announced their own framework based on `graphql-java` that is opinionated towards Spring named [DGS](https://netflix.github.io/dgs/).
They do have some codegen, but I'm unsure it squeezes the most out of the possibilities of schema-first approach :D Plus they seem to be not big fans of required fields which blows my mind a bit as TS clients or any other typed client may want to kill themselves with millions of null checks they have to do :D But required types are supported, but it's just they never use them in the examples and seem to be relying on that quite heavily :) (maybe because in Java everything may become `null` unexpectedly, but nobody cares about Java on Frontend/Client-side I guess, but do care about not doing null checks on every single line :D)

Anyway, just letting you know so maybe you or somebody else will come up with something cool and just so you folks are aware of new player on the JVM market :)

Here is the generator itself.
https://github.com/Netflix/dgs-codegen

**Comment by dotansimha** [comment]
Hi @RIP21 ! Thank you for creating this. I'm moving to discussions since I think it's a better fit for this issue. 

Netflix is doing awesome work with DGS, and I think the DGS plugin is currently very specific to Java/Kotlin and generates output that is very specific for that ecosystem. 
I would love to have plugin for it also in codegen, I think we can introduce some nice concepts for Java ecosystem, but TBH we don't have enough knowledge about the setup of DGS and how to generate code for it.  If someone wants to create one - PRs are always welcome!
